### DISCUSSION TITLE
override captures

### DISCUSSION THREAD

**Discussion by dchinmay2** [nvim-treesitter/nvim-treesitter#2164] [Q&A] [answered]
How can I override captures? When I use `after/queries` my capture shows up in `:TSHighlightCapturesUnderCursor` but the highlighting does not change sometimes

**Comment by stsewd** [comment, marked as answer]
Are you using a custom capture or one listed here https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights?

**Reply by dchinmay2** [reply]
I tried both (`my_custom_comment` and `comment`), neither works

**Reply by stsewd** [reply]
Can you put the exact example/files you are trying with?

**Reply by dchinmay2** [reply]
Fennel:
query:
```scheme
(fn
  name: (symbol) @alabaster.definition
  docstring: (string)* @alabaster.comment)
```
Code:
```fennel
(fn foo []
  "prints foo"
  (print :foo))

(fn bar []
  (print :bar))
```
Syntax tree
```
fn [0, 0] - [2, 15]
  name: symbol [0, 4] - [0, 7]
  parameters [0, 8] - [0, 10]
  docstring: string [1, 2] - [1, 14]
  list [2, 2] - [2, 14]
    symbol [2, 3] - [2, 8]
    string [2, 9] - [2, 13]
fn [4, 0] - [5, 15]
  name: symbol [4, 4] - [4, 7]
  parameters [4, 8] - [4, 10]
  list [5, 2] - [5, 14]
    symbol [5, 3] - [5, 8]
    string [5, 9] - [5, 13]
```

In this example it works but not here: https://github.com/p00f/godbolt.nvim/blob/0be00e2f487daf3a7a2965f2de2f23e9e024eae8/fnl/godbolt/init.fnl#L41

**Reply by dchinmay2** [reply]
oh so it works after i scroll the docstring out of view and scroll it back in again

(the docstring is highlighted as `TSString` first and then with `Comment`) (`alabaster.comment` is `hl_map`ped to `Comment`)

**Reply by dchinmay2** [reply]
It then loses the `Comment` highlight randomly and becomes `String` again

**Reply by stsewd** [reply]
If you are using nightly and compiled with the latest version of trees-sitter, you may be facing https://github.com/nvim-treesitter/nvim-treesitter/issues/1788.

BTW  `docstring: (string)* @alabaster.comment`, the `*` and `+` operators aren't supported on queries at the moment, so the query should be `docstring: (string)? @alabaster.comment`.
